src/utils/buildMetadata.ts

```diff
@@ -166,13 +153,218 @@ export function buildMetadata(sheets: Sheet[], defaultCC: string) {
         optionSets,
         trackedEntityAttributes,
         trackedEntityTypes,
-        //programs,
+        programSections: buildprogramSections(sheets),
+        programs: buildPrograms(sheets),
+        programStages: buildProgramStages(sheets),
+        programStageSections: buildProgramStageSections(sheets),
         programRules: buildProgramRules(sheets),
         programRuleActions: buildProgramRuleActions(sheets),
         programRuleVariables: buildProgramRuleVariables(sheets),
     };
 }
 
+function buildPrograms(sheets: Sheet[]) {
+    const get = (name: string) => getItems(sheets, name);
+
+    const programs = get("programs");
+    const programSectionsData = get("programSections");
+    const programStagesData = get("programStages");
+    const trackedEntityTypes = get("trackedEntityTypes");
+    const categoryCombos = get("categoryCombos");
+
+    return programs.map(program => {
+        const trackedEntityType = {
+            id: getByName(trackedEntityTypes, program.trackedEntityType)?.id
+        };
+
+        const programStages = programStagesData
+            .filter((programStages) => {
+                return programStages?.program === program.name;
+            }).map(({ id }) => ({ id }));
```

Comment:
This is fine, and if you want a less verbose alternative, maybe something like this?

```
const stages = get("programStages");

[...]

const programStages = stages
            .filter(stage => stage.program === program.name)
            .map(stage => ({ id: stage.id }));
```

I find it a bit easier to read this way, but whatever you feel like!

Reply:
Map is definitely more readable, and i see your point about variables names. Adding your suggested version.